docs/content/v0.20/architecture.md

```diff
@@ -1,36 +1,479 @@
 # KubeStellar Architecture
 
-**NOTE**: This is only a very rudimentary start.
+KubeStellar provides multi-cluster deployment of Kubernetes objects, controlled by a simple 
+placement policy, where Kubernetes objects are expressed in their native format with no wrapping
+or bundling. The high-level architecture for KubeStellar is illustrated in Figure 1.
 
-## Higher Level
+<figure>
+  <img src="assets/high-level-architecture.png"  alt="High Level Architecture">
+  <figcaption align="center">Figure 1 - High Level Architecture </figcaption>
+</figure>
 
-At the higher level there are three modules.
+KubeStellar relies on the concept of *spaces*, which are abstractions 
```

Comment:
You may take the definition as we had it in our glossary in KS

Reply:
Do you have a pointer to the glossary ? I could not locate it.

Reply:
https://docs.kubestellar.io/release-0.15/Getting-Started/user-guide/#glossary

Reply:
Thanks, I have updated with that def. Going over comments before updating the PR.